truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/frame/FrameInstance.java

```diff
@@ -47,9 +48,18 @@
     /** @since 0.8 or earlier */
     boolean isVirtualFrame();
 
-    /** @since 0.8 or earlier */
+    /**
+     * @return the node that is calling the next {@link CallTarget target} or <code>null</code> if
+     *         its the topmost frame or the next {@link CallTarget target} was called without a
+     *         {@link TruffleRuntime#createDirectCallNode(CallTarget)} or
+     *         {@link TruffleRuntime#createIndirectCallNode() indirect} call node.
+     * @since 0.8 or earlier
+     **/
     Node getCallNode();
 
-    /** @since 0.8 or earlier */
+    /**
+     * @return the {@link CallTarget target} that created this frame instance by executing it.
```

Comment:
A CallTarget creating a frame seems strange to me (I would think the call node creates the new frame).
Maybe `@return the {@link CallTarget target} that is being executed with this frame.` ?